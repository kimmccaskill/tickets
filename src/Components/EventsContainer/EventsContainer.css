.events-container {
  width: 50em;
}

@media screen and (max-width: 900px) {
  .events-container {
    width: 100%;
  }
}

