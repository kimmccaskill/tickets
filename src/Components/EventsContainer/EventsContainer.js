import React from 'react'
import './EventsContainer.css'
import EventCard from '../../Container/EventCard/EventCard'
import { ClipLoader } from "react-spinners";
import PropTypes from 'prop-types'

export const EventsContainer = ({ events, loader }) => {
  let listOfCards = events.length ? 
  events.sort((a, b) => Date.parse(a.sales.public.startDateTime) - Date.parse(b.sales.public.startDateTime)).map(event => <EventCard key={event.id} {...event}/>) 
  : loader ? <ClipLoader /> : null;
  return (
    <section className='events-container'>
      <h2 className='homepage-title'>Events</h2>
      {listOfCards}
    </section>
  )
}

export default EventsContainer;

EventsContainer.propTypes = {
  events: PropTypes.array,
  loader: PropTypes.any,
}