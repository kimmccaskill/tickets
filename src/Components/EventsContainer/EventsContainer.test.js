import React from 'react';
import EventsContainer from './EventsContainer';
import { shallow } from 'enzyme';
import { mockEvents } from '../../mockData'

describe('EventsContainer', () => {
  let wrapper
  it('should match the snapshot', () => {
    wrapper = shallow(<EventsContainer events={mockEvents}/>);

    expect(wrapper).toMatchSnapshot();
  });
});