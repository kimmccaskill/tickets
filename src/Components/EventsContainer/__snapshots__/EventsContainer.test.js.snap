// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsContainer should match the snapshot 1`] = `ShallowWrapper {}`;
